ddtrace/internal/_encoding.pyx

```diff
@@ -611,6 +684,47 @@ cdef class MsgpackEncoderV04(MsgpackEncoderBase):
                     return ret
         return 0
 
+    cdef inline int _pack_span_events(self, list span_events):
```

Comment:
```suggestion
    cdef inline int _pack_span_events(self, list span_events):
        cdef int ret
        cdef int L
```

Any time you get a chance to predefine values at the top of the function, it'll allow Cython to use the more narrow type (rather than just "PyObject"), and it'll help with type checking and conversions in other parts of the generated code as well.

Reply:
(you should do the same with `attributes`/`new_attributes`/etc/whatever other variables get defined in the function, even if those variables might not get used, Cython always predefines them in the generated code, so doesn't hurt to be explicit with the narrow typing)